specs/WebResourceRequestedWorkers.md

```diff
@@ -0,0 +1,146 @@
+            L"*", COREWEBVIEW2_WEB_RESOURCE_CONTEXT_ALL,
+            COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_SERVICE_WORKER);
+m_webviewEventSource->add_WebResourceRequested(
+        Callback<ICoreWebView2WebResourceRequestedEventHandler>(
+            [this](ICoreWebView2* webview, ICoreWebView2WebResourceRequestedEventArgs* args)
+                -> HRESULT {
+                wil::com_ptr<ICoreWebView2WebResourceRequest> webResourceRequest;
+                CHECK_FAILURE(args->get_Request(&webResourceRequest));
+                std::wstring source;
+                wil::com_ptr<ICoreWebView2WebResourceRequestedEventArgs2>
+                    webResourceRequestArgs;
+                if (SUCCEEDED(args->QueryInterface(IID_PPV_ARGS(&webResourceRequestArgs))))
+                {
+                    COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE requestSource =
+                        COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_ALL;
+                    webResourceRequestArgs->get_RequestSource(&requestSource);
+                    if (requestSource != COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_ALL)
+                    {
+                        source =
+                            L", \"source\": " + WebRequestSourceToString(requestSource);
+                    }
+                }
+                std::wstring message = L"{" + WebResourceRequestedToJsonString(webResourceRequest.get(), source);
+                message += WebViewPropertiesToJsonString(m_webviewEventSource.get());
+                message += L"}";
+                PostEventMessage(message);
+                return S_OK;
+            })
+            .Get(),
+        &m_webResourceRequestedToken);
+```
+
+# API Details
+## IDL
+```c#
+/// Specifies the source of `WebResourceRequested` event.
+[v1_enum]
+typedef enum COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE {
+  /// Indicates that web resource is requested from main page including dedicated workers and iframes.
+  COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_MAIN = 1,
+
+  /// Indicates that web resource is requested from shared worker.
+  COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_SHARED_WORKER = 2,
+
+  /// Indicates that web resource is requested from service worker.
+  COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_SERVICE_WORKER = 4,
+
+  /// Indicates that web resource is requested from any supported source.
+  COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_ALL = 0XFFFFFFFF
+} COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE;
+cpp_quote("DEFINE_ENUM_FLAG_OPERATORS(COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE);")
+
+[uuid(1cc6a402-3724-4e4a-9099-8cf60f2f93a1), object, pointer_default(unique)]
+interface ICoreWebView2: IUnknown {
+  /// Extends AddWebResourceRequestedFilter to support events from shared
+  /// and service workers. To receive all raised events filters have to be added
+  /// before main page navigation.
+  /// Because service worker runs separately from main page its WebResourceRequested
```

Comment:
```suggestion
  /// Because service workers and shared workers run separately from any one HTML document WebResourceRequested
```

Reply:
will fix, but this is applicable only for service workers, not shared